fxgl-media/src/main/kotlin/com/almasb/fxgl/texture/NineSliceTextureBuilder.kt

```diff
@@ -30,202 +31,115 @@ class NineSliceTextureBuilder(val texture: Texture) {
-            val times = h.toInt() / newCenter.height.toInt()
-            val rem = h.toInt() % newCenter.height.toInt()
+        val adjustedCenterBoth= adjustTextureVertically(adjustedCenterHorizontally, h)
 
-            for (i in 1 until times) {
-                newCenter = newCenter.superTexture(center2, VerticalDirection.DOWN)
-            }
+        val newLeft = adjustTextureVertically(left, h)
 
-            if (rem > 0) {
-                newCenter = newCenter.superTexture(center2.subTexture(Rectangle2D(0.0, 0.0, center2.width, rem.toDouble())), VerticalDirection.DOWN)
-            }
-        }
+        val newRight = adjustTextureVertically(right, h)
 
+        val row1 = newLeft.superTexture(adjustedCenterBoth, HorizontalDirection.RIGHT).superTexture(newRight, HorizontalDirection.RIGHT)
 
+        val newBot = adjustTextureHorizontally(bot, w)
 
-        var newLeft = left
-        if (newLeft.height > h) {
-            newLeft = left.subTexture(Rectangle2D(0.0, 0.0, left.width, h))
-        } else if (newLeft.height < h) {
-            val times = h.toInt() / left.height.toInt()
-            val rem = h.toInt() % left.height.toInt()
+        val row2 = botLeft.superTexture(newBot, HorizontalDirection.RIGHT).superTexture(botRight, HorizontalDirection.RIGHT)
 
-            for (i in 1 until times) {
-                newLeft = newLeft.superTexture(left, VerticalDirection.DOWN)
-            }
+        return row0.superTexture(row1, VerticalDirection.DOWN).superTexture(row2, VerticalDirection.DOWN)
+    }
 
-            if (rem > 0) {
-                newLeft = newLeft.superTexture(left.subTexture(Rectangle2D(0.0, 0.0, left.width, rem.toDouble())), VerticalDirection.DOWN)
-            }
-        }
+    private fun adjustTextureVertically(base: Texture, height: Double): Texture {
+        var adjusted = base
 
-        var newRight = right
-        if (newRight.height > h) {
-            newRight = right.subTexture(Rectangle2D(0.0, 0.0, right.width, h))
-        } else if (newRight.height < h) {
-            val times = h.toInt() / right.height.toInt()
-            val rem = h.toInt() % right.height.toInt()
+        if (base.height > height) {
+            adjusted = base.subTexture(Rectangle2D(0.0, 0.0, base.width, height))
+        } else if (base.height < height) {
+            val times = height.toInt() / base.height.toInt()
+            val rem = height.toInt() % base.height.toInt()
 
             for (i in 1 until times) {
-                newRight = newRight.superTexture(right, VerticalDirection.DOWN)
+                adjusted = base.superTexture(base, VerticalDirection.DOWN)
             }
 
             if (rem > 0) {
-                newRight = newRight.superTexture(right.subTexture(Rectangle2D(0.0, 0.0, right.width, rem.toDouble())), VerticalDirection.DOWN)
+                adjusted = base.superTexture(base.subTexture(Rectangle2D(0.0, 0.0, base.width, rem.toDouble())), VerticalDirection.DOWN)
```

Comment:
should this be `adjusted = adjusted.superTexture ... `?

Reply:
Yes, thanks. Fixed!